pkg/sql/sqlstats/insights/integration/integration_test: TestTransactionInsightsFailOnCommit failed

pkg/sql/sqlstats/insights/integration/integration_test.TestTransactionInsightsFailOnCommit [failed](https://mesolite.cluster.engflow.com/invocations/default/44d0bf61-aaf9-40f1-9122-bfdf4c91c9b1?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NxbHN0YXRzL2luc2lnaHRzL2ludGVncmF0aW9uOmludGVncmF0aW9uX3Rlc3Q=) on release-26.1 @ [6810ba76b5dd5adc1c80a7cbe0a270f75d3aab2b](https://github.com/cockroachdb/cockroach/commits/6810ba76b5dd5adc1c80a7cbe0a270f75d3aab2b):


```
=== RUN   TestTransactionInsightsFailOnCommit
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestTransactionInsightsFailOnCommit1948746458
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:182: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    insights_test.go:650: -- test log scope end --
test logs left over in: outputs.zip/logTestTransactionInsightsFailOnCommit1948746458
--- FAIL: TestTransactionInsightsFailOnCommit (249.58s)
=== RUN   TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=false
--- FAIL: TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=false (231.96s)
=== RUN   TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=false/40001_error_exists_in_txn_insights
    insights_test.go:569: condition failed to evaluate within 3m45s: sql: no rows in result set
--- FAIL: TestTransactionInsightsFailOnCommit/enable_recording_SERIALIZATION_CONFLICT_events=false/40001_error_exists_in_txn_insights (225.62s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTransactionInsightsFailOnCommit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63269

--- a/pkg/server/status.go
+++ b/pkg/server/status.go
@@ -444,12 +444,15 @@ func (b *baseStatusServer) localExecutionInsights(
 		insightsCopy := *insight
 		insightsCopy.Statements = make([]*insightspb.Statement, len(insight.Statements))
 		copy(insightsCopy.Statements, insight.Statements)
-		if insight.Transaction != nil && slices.Contains(insight.Transaction.Causes, insightspb.Cause_HighContention) {
-			// Collect high contention insights seperately, they need additional validation / filtering.
-			// If it is valid we will add it to the response later.
+		if insight.Transaction != nil &&
+			slices.Contains(insight.Transaction.Causes, insightspb.Cause_HighContention) &&
+			!slices.Contains(insight.Transaction.Problems, insightspb.Problem_FailedExecution) {
+			// Collect high-contention-only insights separately for
+			// validation against resolved contention events. Insights
+			// that also have FailedExecution are independently valid
+			// and should not be gated on contention resolution.
 			highContentionInsights[insightsCopy.Transaction.ID] = insightsCopy
 		} else {
-			// All other insights are included in the response.
 			response.Insights = append(response.Insights, insightsCopy)
 		}
 	})

